src/Compilers/CSharp/Portable/Symbols/Source/SourceOrdinaryMethodSymbol.cs

```diff
@@ -1120,7 +1120,6 @@ private static bool HaveSameConstraints(SourceOrdinaryMethodSymbol part1, Source
                 return true;
             }
 
-            Debug.Assert(part1.NonNullTypes == part2.NonNullTypes);
```

Comment:
A design question: it feels like we should warn or error if the NonNullTypes for a partial method differs.

Reply:
> A design question: it feels like we should warn or error if the NonNullTypes for a partial method differs.

I do not think it is necessary (especially with syntax based approach) as long as we warn in case of mismatch in nullability of types between both declarations. I believe we do have this check in place already.

---
In reply to: [224183709](https://github.com/dotnet/roslyn/pull/30387#discussion_r224183709) [](ancestors = 224183709)

Reply:
Note, this is just the first step, at the end symbols won't have NonNullTypes context associated with them, there won't be a property like this on Symbol

---
In reply to: [224243539](https://github.com/dotnet/roslyn/pull/30387#discussion_r224243539) [](ancestors = 224243539,224183709)